pypeit/flatfield.py

```diff
@@ -1062,14 +1066,17 @@ def fit(self, spat_illum_only=False, debug=False):
                                         * np.fmax(spec_model[onslit_tweak], 1.0)
 
             # Construct the pixel flat
-            #self.mspixelflat[onslit] = rawflat[onslit]/self.flat_model[onslit]
-            #self.mspixelflat[onslit_tweak] = 1.
             #trimmed_slitid_img_anew = self.slits.slit_img(pad=-trim, slitidx=slit_idx)
             #onslit_trimmed_anew = trimmed_slitid_img_anew == slit_spat
             self.mspixelflat[onslit_tweak] = rawflat[onslit_tweak]/self.flat_model[onslit_tweak]
             # TODO: Add some code here to treat the edges and places where fits
             #  go bad?
 
+            # Minimum wavelength?
```

Comment:
@profxj my suggestion here is to adopt the same procedure I adopted for instrument specific tweaks to standard stars, i.e. 

self.mispixelflat = spectrograph.tweak_pixelflat(self.mispixelflat, waveimg)

Or something like that. The point is that you can imagine cases where the user wants to perform instrument specific tweaks to the flat besides just masking as a function of wavelength. The Box B&C case that came up in the user slack is one example (although there it might be a straight wavelength tweak as well). The point is that then that code can just go in the spectrograph class rather than here. I guess it is a toss up where to put this code, but in that way users could for example deal with scattered light issues as well. 

Also, I think you should change pixelflat_min_wave to pixelflat_min_max_wave and make it a tuple to deal with both cases of minimum and maximum tweaks.

Reply:
Added `pixelflat_max_wave` as a separate parameter.
Tuple's are a pain.  All the more so if one entry can be `None`.

Not moving to `spectrograph`